modules/create_timepoint/php/ajaxtimepoint.class.inc

```diff
@@ -0,0 +1,292 @@
+<?php
+/**
+ * POST request methods.
+ *
+ * Used to retrieve info & create timepoint by the create_timepoint form.
+ * Ensures the user can create timepoint before processing
+ * the $_POST (array) data.
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Create_Timepoint
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\create_timepoint;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * The NDB_Page for the POST requests.
+ *
+ * @category Loris
+ * @package  Create_Timepoint
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class AjaxTimepoint extends \NDB_Page
+{
+    public $skipTemplate = true;
+
+    /**
+     * This function will return a json file with DCCID and PSCID
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     * @throws \DatabaseException
+     * @throws \DeprecatedException
+     * @throws \LorisException
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        /**
+         * Ensure it's a POST request.
+         */
+        if ($request->getMethod() != 'POST') {
+            return (new \LORIS\Http\Response())
+                ->withHeader('Content-Type', 'application/json')
+                ->withStatus(405)
+                ->withHeader('Allow', 'POST')
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        'Only POST requests are allowed'
+                    )
+                );
+        }
+
+        /**
+         * Parse POST request body.
+         */
+        $_POST = $request->getParsedBody();
```

Comment:
This goes against the purpose of using the PSR7 implementations. You'd be better off accessing these vairables using `$request->getAttribute()`.

Reply:
Hey @johnsaigle I messaged you on slack about this and maybe you haven't seen it yet. I'm confused by your comment because the getParsedBody() is from the the ServerRequestInterface and I think is part of the PSR7 interface. The getParsedBody() returns an array of what is sent in the POST javascript fetch method and pertaining the the "body" key.

Reply:
The `$request->getParsedBody` is fine, I think it's the `$_POST` he has a problem with since it's a special PHP superglobal name. I think it should be fine (and less confusing) if you just give the variable a different name.

Reply:
Thank you @driusan, that makes sense. I just copied the old code from the previous commits and which is why the variable is named $_POST. I'll change it to something else.